### PR TITLE
Update `fetch_data`

### PR DESCRIPTION
**Pull request by domesticmouse**
Fixes https://github.com/flutter/website/issues/9826

### CODE REVIEW

File: examples/cookbook/networking/fetch_data/lib/main.dart [hunk 1]
```diff
@@ -36,11 +36,19 @@ class Album {
...
+        'userId': int userId,
+        'id': int id,
+        'title': String title,
+      } =>
+        Album(
+          userId: userId,
+          id: id,
+          title: title,
+        ),
+      _ => throw 'Failed to load album',
```
Comment (parlough) [inline examples/cookbook/networking/fetch_data/lib/main.dart:50]: [`FormatException`](https://api.dart.dev/stable/dart-core/FormatException-class.html) is made for this situation.

```suggestion
      _ => throw FormatException('Failed to load album.'),
```

However I expected this to trigger a lint. What do you think about enabling [`only_throw_errors`](https://dart.dev/lints/only_throw_errors)?

File: src/cookbook/networking/fetch-data.md [hunk 2]
```diff
@@ -76,7 +76,7 @@ First, create an `Album` class that contains the data from the
 network request. It includes a factory constructor that
 creates an `Album` from JSON.
 
-Converting JSON by hand is only one option.
+Converting JSON using pattern matching is only one option.
```
Comment (parlough) [inline src/cookbook/networking/fetch-data.md:79]: Can you cross link "pattern matching" to https://dart.dev/language/patterns?

Some readers may not be familiar yet.

**Review by parlough — APPROVED** [review]
Thanks @domesticmouse! Would you be able to make similar changes to the other recipes in this directory? Many of them mention they depend on the code from this one. It should mostly be updating their `fromJson` implementations to match.

File: src/cookbook/networking/fetch-data.md [hunk 3]
```diff
@@ -344,4 +360,5 @@ class _MyAppState extends State<MyApp> {
 [`initState()`]: {{site.api}}/flutter/widgets/State/initState.html
 [Mock dependencies using Mockito]: {{site.url}}/cookbook/testing/unit/mocking
 [JSON and serialization]: {{site.url}}/data-and-backend/serialization/json
+[pattern matching]: https://dart.dev/language/patterns
```
Comment (domesticmouse) [inline src/cookbook/networking/fetch-data.md:363]: I'm guessing https://dart.dev has a template variable?

File: src/cookbook/networking/fetch-data.md (hunk 2 above)
Comment (domesticmouse) [reply src/cookbook/networking/fetch-data.md]: done

File: src/cookbook/networking/fetch-data.md (hunk 3 above)
Comment (domesticmouse) [reply src/cookbook/networking/fetch-data.md]: Found it

**Review by parlough — APPROVED** [review]
That ended up touching more areas than I expected, thanks for all of that!! Looks great :D

**Pull request merged**